q2_diversity/_filter.py

```diff
@@ -12,13 +12,16 @@
 
 def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
                            metadata: qiime2.Metadata,
-                           where: str=None) -> skbio.DistanceMatrix:
+                           where: str=None,
+                           exclude_ids: bool=False) -> skbio.DistanceMatrix:
     ids_to_keep = metadata.ids(where=where)
     # NOTE: there is no guaranteed ordering to output distance matrix because
     # `ids_to_keep` is a set, and `DistanceMatrix.filter` uses its iteration
     # order.
     try:
+        if exclude_ids:
```

Comment:
This is a bit on the stylistic side, but I think it makes sense to move this check up to line 18 above. I suggest this because this code (`ids_to_keep = ...`) doesn't make much sense to include in the `try/except` block, and in particular, you aren't checking for any potential errors that **could** be raised in L23 (I'm not suggesting you do). Make sense?